lib/class-wp-theme-json-resolver.php

```diff
@@ -270,6 +274,10 @@ private static function get_user_origin() {
 	 * @return WP_Theme_JSON
 	 */
 	public function get_origin( $theme_support_data = array(), $origin = 'user', $merged = true ) {
+		if ( ! gutenberg_experimental_global_styles_has_theme_json_support() ) {
```

Comment:
I'm not sure if should be this function verifying if theme.json is supported or not. If this function is called with an origin user  I guess it should return the origin user. The caller of this function should probably be the one making the checks?